src/components/Slideshow.tsx

```diff
@@ -0,0 +1,97 @@
+import { Body, Box, Heading, Image, SubHeading } from '@mycrypto/ui';
+import type { FunctionComponent } from 'react';
+import { useEffect, useState } from 'react';
+
+export interface Slide {
+  title: string;
+  text: string;
+  image: string;
+}
+
+export interface SlideshowButtonProps {
+  index: number;
+  title: string;
+  isActive: boolean;
+
+  onClick(index: number): void;
+}
+
+export interface SlideshowProps {
+  title: string;
+  imagePosition?: 'left' | 'right';
+  slides: Slide[];
+}
+
+export const SlideshowButton: FunctionComponent<SlideshowButtonProps> = ({
+  title,
+  isActive,
+  onClick,
+  index
+}) => {
+  const handleClick = () => {
+    onClick(index);
+  };
+
+  return (
+    <SubHeading
+      marginRight="4"
+      variant={isActive ? 'variants.slideshow.active' : 'variants.slideshow.inactive'}
+      onClick={handleClick}
+    >
+      {title}
+    </SubHeading>
+  );
+};
+
+export const Slideshow: FunctionComponent<SlideshowProps> = ({
+  title,
+  slides,
+  imagePosition = 'left'
+}) => {
+  const [slide, setSlide] = useState(0);
+
+  useEffect(() => {
+    const timeout = setTimeout(() => {
+      if (slides.length - 1 > slide) {
+        return setSlide((state) => state + 1);
+      }
+
+      setSlide(0);
+    }, 7500);
+
+    return () => clearTimeout(timeout);
+  }, [slide]);
+
+  const handleClick = (index: number) => setSlide(index);
```

Comment:
Looks like this clears the timeout, can you confirm?

Reply:
Yes it does, as intended.